api/src/repositories/submission-repository.ts

```diff
@@ -1454,6 +1465,88 @@ export class SubmissionRepository extends BaseRepository {
     return response.rows[0];
   }
 
+  /**
+   * Get a submission feature record by uuid.
+   *
+   * @param {string} submissionUuid
+   * @return {*}  {Promise<SubmissionFeatureRecord>}
+   * @memberof SubmissionRepository
+   */
+  async getSubmissionFeatureByUuid(submissionUuid: string): Promise<SubmissionFeatureRecord> {
+    const sqlStatement = SQL`
+      SELECT
+        *
+      FROM
+        submission_feature
+      WHERE
+        uuid = ${submissionUuid};
+    `;
+
+    const response = await this.connection.sql(sqlStatement, SubmissionFeatureRecord);
+
+    if (response.rowCount !== 1) {
+      throw new ApiExecuteSQLError('Failed to get submission feature record', [
+        'SubmissionRepository->getSubmissionFeatureByUuid',
+        `rowCount was ${response.rowCount}, expected rowCount === 1`
+      ]);
+    }
+
+    return response.rows[0];
+  }
+
+  /**
+   * Find and return all submission feature records that match the provided criteria.
+   *
+   * @param {{
+   *     submissionId?: number;
+   *     systemUserId?: number;
+   *     featureTypeNames?: string[];
+   *     includeDeleted?: boolean;
+   *   }} [criteria]
+   * @return {*}  {Promise<SubmissionFeatureRecord[]>}
+   * @memberof SubmissionRepository
+   */
+  async findSubmissionFeatures(criteria?: {
+    submissionId?: number;
+    systemUserId?: number;
+    featureTypeNames?: string[];
+    includeDeleted?: boolean;
+  }): Promise<SubmissionFeatureRecord[]> {
+    const knex = getKnex();
+
+    const queryBuilder = knex.queryBuilder();
+
+    queryBuilder.select().from('submission_feature');
+
+    if (criteria?.submissionId) {
+      // Filter by submitter system user id
+      queryBuilder.where('submission_id', criteria.submissionId);
+    }
+
+    if (criteria?.systemUserId) {
+      // Filter by submitter system user id
+      queryBuilder.where('systemUserId', criteria.systemUserId);
+    }
+
+    if (criteria?.featureTypeNames?.length) {
+      const featureTypeNames = criteria?.featureTypeNames;
+      // Filter by feature type names
+      queryBuilder.whereIn('feature_type_id', (qb) => {
+        qb.select('feature_type_id')
+          .from('feature_type')
+          .where(
+            knex.raw('LOWER(name)'),
+            'IN',
+            featureTypeNames.map((featureTypeName) => featureTypeName.toLowerCase())
```

Comment:
Is this syntactically correct? I thought you had to wrap the `WHERE IN` values in brackets, e.g. `WHERE LOWER(name) IN (?,?,?)`

Reply:
On second thought, I'm guessing this is handled by Knex 🤔

Reply:
Sorry, yeah I believe knex is cool with it. It did take a few trys to get it right though. I tried using `whereIn` which you would think would work exactly like the current implementation, just minus the `IN` param. But the type checking kept throwing errors. Changing it to a regular `Where` + the `IN` param made it happy.